backend/worker/error_scorer.go

```diff
@@ -0,0 +1,283 @@
+package worker
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/highlight-run/highlight/backend/model"
+	"github.com/highlight-run/highlight/backend/redis"
+	"github.com/highlight-run/highlight/backend/store"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type ErrorGroupWithLastObjectId struct {
+	ID                int
+	ProjectID         int
+	CreatedAt         time.Time
+	LastErrorObjectId int
+}
+
+type ErrorScorer struct {
+	store *store.Store
+	db    *gorm.DB
+	redis *redis.Client
+}
+
+func NewErrorScorer(store *store.Store, db *gorm.DB, redis *redis.Client) *ErrorScorer {
+	return &ErrorScorer{
+		store: store,
+		db:    db,
+		redis: redis,
+	}
+}
+
+func (errorScorer *ErrorScorer) ScoreImpactfulErrors(ctx context.Context) {
+	var errorGroups []*ErrorGroupWithLastObjectId
+
+	lastComputedErrorInstanceId, _ := errorScorer.redis.GetLastComputedImpactfulErrorObjectId(ctx)
+
+	if err := errorScorer.db.Raw(`
+		SELECT grp.id, grp.created_at, grp.project_id, MAX(obj.id) as last_error_object_id
```

Comment:
This query uses the indices for scanning, and looks like the greatest cost is on the grouping and sorting

```
-[ RECORD 1 ]-------------------------
QUERY PLAN | Sort  (cost=1733.59..1734.07 rows=195 width=24) (actual time=3.998..4.005 rows=136 loops=1)
-[ RECORD 2 ]-------------------------
QUERY PLAN |   Sort Key: (max(obj.id))
-[ RECORD 3 ]-------------------------
QUERY PLAN |   Sort Method: quicksort  Memory: 34kB
-[ RECORD 4 ]-------------------------
QUERY PLAN |   ->  GroupAggregate  (cost=1722.76..1726.17 rows=195 width=24) (actual time=3.763..3.959 rows=136 loops=1)
-[ RECORD 5 ]-------------------------
QUERY PLAN |         Group Key: grp.id
-[ RECORD 6 ]-------------------------
QUERY PLAN |         ->  Sort  (cost=1722.76..1723.24 rows=195 width=24) (actual time=3.757..3.828 rows=1067 loops=1)
-[ RECORD 7 ]-------------------------
QUERY PLAN |               Sort Key: grp.id
-[ RECORD 8 ]-------------------------
QUERY PLAN |               Sort Method: quicksort  Memory: 124kB
-[ RECORD 9 ]-------------------------
QUERY PLAN |               ->  Nested Loop  (cost=1.15..1715.34 rows=195 width=24) (actual time=0.280..3.532 rows=1067 loops=1)
-[ RECORD 10 ]-------------------------
QUERY PLAN |                     ->  Index Scan using idx_error_objects_created_at on error_objects obj  (cost=0.57..53.96 rows=195 width=8) (actual time=0.039..1.617 rows=1067 loops=1)
-[ RECORD 11 ]-------------------------
QUERY PLAN |                           Index Cond: (created_at >= (now() - '00:10:00'::interval))
-[ RECORD 12 ]-------------------------
QUERY PLAN |                           Filter: (id > 0)
-[ RECORD 13 ]-------------------------
QUERY PLAN |                     ->  Memoize  (cost=0.57..8.59 rows=1 width=20) (actual time=0.002..0.002 rows=1 loops=1067)
-[ RECORD 14 ]-------------------------
QUERY PLAN |                           Cache Key: obj.error_group_id
-[ RECORD 15 ]-------------------------
QUERY PLAN |                           Cache Mode: logical
-[ RECORD 16 ]-------------------------
QUERY PLAN |                           Hits: 931  Misses: 136  Evictions: 0  Overflows: 0  Memory Usage: 17kB
-[ RECORD 17 ]-------------------------
QUERY PLAN |                           ->  Index Scan using error_groups_pkey on error_groups grp  (cost=0.56..8.58 rows=1 width=20) (actual time=0.010..0.010 rows=1 loops=136)
-[ RECORD 18 ]-------------------------
QUERY PLAN |                                 Index Cond: (id = obj.error_group_id)
-[ RECORD 19 ]-------------------------
QUERY PLAN | Planning Time: 0.352 ms
-[ RECORD 20 ]-------------------------
QUERY PLAN | Execution Time: 4.053 ms
EXPLAIN
```